metricflow/dataflow/builder/node_evaluator.py

```diff
@@ -92,13 +98,13 @@ def join_description(self) -> JoinDescription:
             ]
         )
 
-        include_specs.extend(
-            [
-                LinklessEntitySpec.from_reference(spec.entity_links[0])
-                for spec in self.satisfiable_linkable_specs
-                if len(spec.entity_links) > 0
-            ]
-        )
+        # Get the specs needed to join onto this node.
+        if self.node_to_join.aggregated_to_elements:
+            include_specs.extend(self.node_to_join.aggregated_to_elements)
```

Comment:
Did we want these as-is or did we want to convert them to the LinklessEntitySpec types at this point?

Reply:
We want them as-is. The reason they were always `LinklessEntitySpecs` previously is that we were always joining to source nodes (which would not have joins contained WITHIN them). In this case, we are joining to a node that has joins within it, so its instance specs might have entity links.